### ISSUE TITLE
[iOS] "Forget by default" feature

### ISSUE THREAD

**Issue by cuba** [brave/brave-browser#36406]
### Description: 
Create a new option for cookie handling in shields, where first-party data was cleared when you navigate away from a site.  In addition to the new shields setting, this would also have new UI (nav bar button) for indicating that you want the site to persist storage.

Full specs can be found here:
https://docs.google.com/document/d/1TeDqwnKqOJqSv0b80liZjAFR1BUPc6urGkkHiHN526s/edit#

**Comment by ShivanKaul** [comment]
From @cuba, this is what we can use to clear data:

```
/** @constant WKWebsiteDataTypeFetchCache On-disk Fetch caches. */
@available(iOS 11.3, *)
public let WKWebsiteDataTypeFetchCache: String

/** @constant WKWebsiteDataTypeDiskCache On-disk caches. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeDiskCache: String

/** @constant WKWebsiteDataTypeMemoryCache In-memory caches. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeMemoryCache: String

/** @constant WKWebsiteDataTypeOfflineWebApplicationCache HTML offline web application caches. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeOfflineWebApplicationCache: String

/** @constant WKWebsiteDataTypeCookies Cookies. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeCookies: String

/** @constant WKWebsiteDataTypeSessionStorage HTML session storage. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeSessionStorage: String

/** @constant WKWebsiteDataTypeLocalStorage HTML local storage. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeLocalStorage: String

/** @constant WKWebsiteDataTypeWebSQLDatabases WebSQL databases. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeWebSQLDatabases: String

/** @constant WKWebsiteDataTypeIndexedDBDatabases IndexedDB databases. */
@available(iOS 9.0, *)
public let WKWebsiteDataTypeIndexedDBDatabases: String

/** @constant WKWebsiteDataTypeServiceWorkerRegistrations Service worker registrations. */
@available(iOS 11.3, *)
public let WKWebsiteDataTypeServiceWorkerRegistrations: String

/** @constant WKWebsiteDataTypeFileSystem File system storage. */
@available(iOS 16.0, *)
public let WKWebsiteDataTypeFileSystem: String

/** @constant WKWebsiteDataTypeSearchFieldRecentSearches Search field history. */
@available(iOS 17.0, *)
public let WKWebsiteDataTypeSearchFieldRecentSearches: String

/** @constant WKWebsiteDataTypeMediaKeys MediaKeys storage */
@available(iOS 17.0, *)
public let WKWebsiteDataTypeMediaKeys: String

/** @constant WKWebsiteDataTypeHashSalt Hash salt for deviceId */
@available(iOS 17.0, *)
public let WKWebsiteDataTypeHashSalt: String
```

**Comment by cuba** [comment]
Replaced by Shred feature
